relayer/relayer/src/contracts/azero.rs

```diff
@@ -35,9 +38,6 @@ pub enum AzeroContractError {
 
     #[error("Missing or invalid field")]
     MissingOrInvalidField(String),
-
-    #[error("Dry run failed")]
-    DryRunFailed(String),
```

Comment:
It actually would be best to dry-run here, so we avoid generating additional costs when crash-looping (also ensure that we dry run when submitting to Ethereum).

Reply:
AFAIK dry running was not used anywhere but in `is_halted` for some reason, which is a query and therefore does not cost gas.  Should we want to start dry running this would have to be written from scratch for request signing, which is not a bad idea I guess. Question is if we want to do it in this PR or in a follow up?

Reply:
We can do it in the follow-up if you prefer that

Reply:
yeah, I doubt I will have the time. It will be pointed out in the audit though, so we should probably mark it as out-of-scope as known. But if they see other gas savings this is within the scope